src/pages/index.js

```diff
@@ -0,0 +1,363 @@
+const previewModalCaptionEl = document.querySelector(".modal__caption");
+const previewModalCloseBtn = previewModal.querySelector(".modal__close-btn");
+
+// card related elements
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+let selectedCard;
+let selectedCardId;
+
+function handleDeleteSumbit(evt) {
+  evt.preventDefault();
+  const button = evt.target.querySelector(".modal__submit-btn_delete");
+  api.deleteCard(selectedCardId).then(() => {
+    selectedCard.remove();
+    closeModal(deleteModal);
+  });
+}
+
+function handleDeleteCard(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(deleteModal);
+}
+
+// what is the event
+// and what do I want to happen which mean use or call teh our function we created
+
+function handleLike(evt, cardData) {
+  const isLiked = evt.classList.contains("card__like-btn_active");
+  console.log("Is card liked", isLiked);
+  api
+    .changeLikeStatus(cardData._id, isLiked)
+    .then(() => {
+      evt.classList.toggle("card__like-btn_active");
+    })
+    .catch(console.error);
+
+  // send a request to the API (call changeCardLike)
+  // then -> change the like status (toggle the class)
+}
+
+function handleEscape(evt) {
+  if (evt.key === "Escape") {
+    const activeModal = document.querySelector(".modal_opened");
+    console.log("escape clicked");
+    closeModal(activeModal);
+  }
+}
+
+function openModal(modal) {
+  console.log(modal);
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscape);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscape);
+}
+
+editProfileModal.addEventListener("mousedown", (evt) => {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(editProfileModal);
+  }
+});
+
+avatarModal.addEventListener("mousedown", (evt) => {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(avatarModal);
+  }
+});
+
+cardSubmitModal.addEventListener("mousedown", (evt) => {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(cardSubmitModal);
+  }
+});
+
+previewModal.addEventListener("mousedown", (evt) => {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(previewModal);
+  }
+});
+
+function setUserData(data) {
+  profileTitle.textContent = data.name;
+  profileDescription.textContent = data.about;
+  profileAvatar.src = data.avatar;
+}
+
+function handleEditFormSubmit(evt) {
+  console.log("submit it");
+  evt.preventDefault();
+  editButtonEl.textContent = "Loading...";
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      setUserData(data);
+    })
+    .catch(console.error)
+
+    .finally(() => {
+      editButtonEl.textContent = "Save";
+    });
+
+  closeModal(editProfileModal);
```

Comment:
`  closeModal(editProfileModal);
`

This should be on line 213 in `.then`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**